Content.Client/_White/ItemSlotRenderer/ItemSlotRendererSystem.cs

```diff
@@ -0,0 +1,143 @@
+using Robust.Shared.Timing;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Content.Client._White.ItemSlotRenderer;
+
+/// <summary>
+/// I can feel my grip on reality slowly slipping.
+/// </summary>
+public sealed class ItemSlotRendererSystem : EntitySystem
+{
+    [Dependency] private readonly IReflectionManager _reflection = default!;
+    [Dependency] private readonly ItemSlotsSystem _slot = default!;
+    [Dependency] private readonly IClyde _clyde = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ItemSlotRendererComponent, ComponentStartup>(OnStartup);
+        SubscribeLocalEvent<ItemSlotRendererComponent, ComponentRemove>(OnRemove);
+        SubscribeLocalEvent<ItemSlotRendererComponent, EntInsertedIntoContainerMessage>(OnInsertIntoContainer);
+        SubscribeLocalEvent<ItemSlotRendererComponent, EntRemovedFromContainerMessage>(OnRemoveFromContainer);
+
+    }
+    private void OnInsertIntoContainer(EntityUid uid, ItemSlotRendererComponent comp, EntInsertedIntoContainerMessage args)
+    {
+        if (args.Container is not ContainerSlot || !_timing.IsFirstTimePredicted)
+            return;
+
+        comp.CachedEntities[args.Container.ID] = args.Entity;
+    }
+
+    private void OnRemoveFromContainer(EntityUid uid, ItemSlotRendererComponent comp, EntRemovedFromContainerMessage args)
+    {
+        if (args.Container is not ContainerSlot || !_timing.IsFirstTimePredicted)
+            return;
+
+        comp.CachedEntities[args.Container.ID] = null;
+    }
+
+    private void OnRemove(EntityUid uid, ItemSlotRendererComponent comp, ComponentRemove args)
+    {
+        foreach (var (_, renderTexture) in comp.CachedRT)
+            renderTexture.Dispose();
+    }
+
+    private void OnStartup(EntityUid uid, ItemSlotRendererComponent comp, ComponentStartup args)
+    {
+        if(!TryComp<SpriteComponent>(uid, out var sprite))
+        {
+            Log.Error($"ItemSlotRendererCompontn requires SpriteComponent to work, but {ToPrettyString(uid)} did not have one. Removing ItemSlotRenderer.");
+            RemComp<ItemSlotRendererComponent>(uid);
+            return;
+        }
+
+        foreach (var kvp in comp.PrototypeLayerMappings)
+        {
+
+            (string slotId, object mapKey) = kvp;
+            bool isEnum = false;
+            if (_reflection.TryParseEnumReference((string)mapKey, out var e))
+            {
+                mapKey = e;
+                isEnum = true;
+            }
+            if (!sprite.LayerMapTryGet(mapKey, out _) && comp.ErrorOnMissing)
+            {
+                Log.Warning($"ItemSlotRenderer: Tried to add a missing layer under the {(isEnum ? "enum" : "string")} key {mapKey}. Skipping missing layer. If this is unwanted, set component's AddMissingLayers to true.");
```

Comment:
also yeah, i forgot to update this error message when i renamed `AddMissingLayers` to `ErrorOnMissing`
Actually, that bool could probably be removed altogether - I can't come up with a good use case for it.